public/pages/Rules/utils/helpers.tsx

```diff
@@ -177,3 +177,19 @@ export function setBreadCrumb(
 ) {
   breadCrumbSetter?.([BREADCRUMBS.SECURITY_ANALYTICS, BREADCRUMBS.RULES, breadCrumb]);
 }
+
+export function getLogTypeFromLogSource(logSource: { [k: string]: string }) {
+  const logTypes = new Set(ruleTypes.map(({ value }) => value));
+
+  let logType;
+  ['product', 'category', 'service'].some((key) => {
```

Comment:
Nitpick: Not blocking, but if you end up revising this PR in some way, it would be nice for `'product', 'category', 'service'` to be a constant somewhere with a comment explaining what they indicate.

Reply:
Makes sense, let me do it now

Reply:
yeah or we can iterate through keys for better readability to find a matching log type, something like this..

```
for (const key of ['product', 'category', 'service']) {
```